res/enkf/es_update.py

```diff
@@ -1,4 +1,108 @@
+import logging
 from res import _lib
+from res.enkf.enums import RealizationStateEnum
+from res.analysis.enums import AnalysisModuleOptionsEnum
+
+
+logger = logging.getLogger(__name__)
+
+
+def size_is_big_enough(active_ens_size, required_ens_size):
+    if active_ens_size < required_ens_size:
+        logger.error(
+            "** ERROR ** There are %d active realisations left, which is "
+            "less than the minimum specified - stopping assimilation.",
+            active_ens_size,
+        )
+        return False
+    return True
+
+
+def config_is_correct(updatestep, iterable):
+    # exit if multi step update with iterable modules
+    if len(updatestep) > 1 and iterable:
+        raise ValueError(
+            "** ERROR: Can not combine iterable modules with multi step "
+            "updates - sorry"
+        )
+    return True
+
+
+def analysis_smoother_update(
+    updatestep,
+    total_ens_size,
+    obs,
+    shared_rng,
+    analysis_config,
+    ensemble_config,
+    source_fs,
+    target_fs,
+):
+    source_state_map = source_fs.getStateMap()
+
+    ens_mask = source_state_map.selectMatching(RealizationStateEnum.STATE_HAS_DATA)
+    iens_active_index = [i for i in range(len(ens_mask)) if ens_mask[i]]
+    module = analysis_config.getActiveModule()
+
+    if not size_is_big_enough(
+        len(iens_active_index),
+        min(analysis_config.minimum_required_realizations, total_ens_size),
+    ) or not config_is_correct(
+        updatestep,
+        module.checkOption(AnalysisModuleOptionsEnum.ANALYSIS_ITERABLE),
+    ):
+        return False
```

Comment:
Perhaps this should raise a custom exception with some context? Catch it on the other side and give some feedback?

Reply:
I was just replication behaviour, but I do think it is a bit odd... I think returning  False should be sufficient

Reply:
and just log the error message

Reply:
Maybe, though this is what happens on the other side:
```
        if not success:
            raise ErtRunError("Analysis of simulation failed!")
```
which is not very helpful, however I see your argument that we should replicate behavior in this PR, could you create an issue for better feedback?

Reply:
https://github.com/equinor/ert/issues/3008